### ISSUE TITLE
Add a new `brew maintainer` command for project administration

### ISSUE THREAD

**Issue by issyl0** [Homebrew/brew#15151]
### Verification

- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### Provide a detailed description of the proposed feature

Make a new `brew maintainers` command (or something, naming is hard) to enable the PL/PLC to add a new maintainer. Some ideas for functionality:

- Adds a new person to the GitHub org.
- Adds a new person to the `members` team (so they can read governance documents).
- Adds a new person to the `maintainers` team.
- Has a switch to enable the standard maintainer stipend (Sponsors GraphQL API). Maybe it should calculate the >=30 days of maintainer-y-ness and doesn't allow doing it before then, since humans are fallible and calendars are difficult.

### What is the motivation for the feature?

Homebrew is growing. We're bringing on new maintainers reasonably regularly these days - fantastic! But the [current process of giving people access to things](https://docs.brew.sh/New-Maintainer-Checklist) is manual, the doc is hard to find and so some of the steps get forgotten.

Having a reasonably automated "<person> is a new maintainer on "core", set up all the GitHub-related things" would be good. Especially since we only have a few org owners now - this is a good thing but increases the support burden on those three people.

### How will the feature be relevant to at least 90% of Homebrew users?

It won't, but it will make project admin less susceptible to human error or forgetfulness.

### What alternatives to the feature have been considered?

Organization owners continuing to click manually in the GitHub UIs.

**Comment by MikeMcQuaid** [comment]
Sorry @issyl0: I'm not sure we want a command for this:
- literally only ~3 people will have the privileges to run it and do all these things
- I don't think it's worth having building and maintaining a command given the above
- it's not worth the security implications to have a token that can do all of this

Instead I'd suggest something like a workflow that can, given a maintainer name, open an "onboarding issue" in a private repository with steps that the maintainer (or their "onboarding buddy"?) can check off as we work our way through the steps.

**Comment by issyl0** [comment]
> literally only ~3 people will have the privileges to run it and do all these things

Yeah. But getting Slack pings that we've forgotten to add people to teams, and the amount of clicking required to fix it, is annoying.

> it's not worth the security implications to have a token that can do all of this

Org owners can make tokens that can do all of this anyway. Not making this command will only limit or slow down the damage if one of those were to be compromised. That in itself is a good thing.

**Comment by MikeMcQuaid** [comment]
> Org owners can make tokens that can do all of this anyway.

They can but it's not clear they do and I'd really rather they didn't. When we can enforce that they don't: we will.

---

I agree there's a problem worth solving here, I just don't think the best solution to that problem is us adding a new Homebrew command that will be very infrequently used by very few people.
